lib/cadet/assessments/submission.ex

```diff
@@ -20,26 +20,51 @@ defmodule Cadet.Assessments.Submission do
 
     belongs_to(:assessment, Assessment)
     belongs_to(:student, CourseRegistration)
+    belongs_to(:team, Team)
     belongs_to(:unsubmitted_by, CourseRegistration)
-    has_many(:answers, Answer)
+
+    has_many(:answers, Answer, on_delete: :delete_all)
```

Comment:
Interesting that we never had this `on_delete` before. What was the existing submission delete behaviour then?

Reply:
Before the implementation of Team Submission, there was only Individual Submission. Thus there was no need to delete any submissions. With the introduction of teams and team formation, we can delete an existing team and the submission that is tagged to the team. Thus, we introduced deleting submission here, along with the answers tagged to it.